### PR TITLE
Animal model description in module_overview.md

### PR DESCRIPTION
**Pull request by TaranRallings**
# Description

This is a simple documentation update to finally add a description of the animal module. It you think I'm missing important content or that my explanation is bananas, let me know and I'll add/rework.

Fixes # (issue)

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Key checklist

- [x] Make sure you've run the `pre-commit` checks: `$ pre-commit run -a`
- [x] All tests pass: `$ poetry run pytest`

## Further checks

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works
- [ ] Relevant documentation reviewed and updated

### CODE REVIEW

**Review by jacobcook1995 — APPROVED** [review]
Looks sensible to me!

File: docs/source/virtual_ecosystem/module_overview.md [hunk 1]
```diff
@@ -212,6 +212,47 @@ Further theoretical background for the Soil Model can be found
 
 ## Animal Model
 
+The Animal Model simulates the animal consumers for the Virtual Ecosystem. We follow the
+Madingley Model {cite}`harfoot_madingley_2014` to provide the foundational structure
+as well as some of the dynamics. The key processes of which are:
```
Comment (vgro) [inline docs/source/virtual_ecosystem/module_overview.md:217]:
```suggestion
as well as some of the dynamics. The key processes of the model are:
```
not sure what you meant by 'which' here, maybe 'the model'?

File: docs/source/virtual_ecosystem/module_overview.md [hunk 2]
```diff
@@ -212,6 +212,47 @@ Further theoretical background for the Soil Model can be found
...
+- migration
+- birth
+- metamorphosis
+- metabolism
+- natural mortality
+
+### Functional Groups
+
+Animals within the Animal Model are sorted into functional groups, not biological
+species. Functional groups share functional traits and body-mass ranges in common and
```
Comment (vgro) [inline docs/source/virtual_ecosystem/module_overview.md:229]:
```suggestion
species. Functional groups share functional traits and body-mass ranges and
```

File: docs/source/virtual_ecosystem/module_overview.md [hunk 3]
```diff
@@ -212,6 +212,47 @@ Further theoretical background for the Soil Model can be found
...
+so behave similarly within the ecosystem. Defining a functional group within the
+animal model requires the following traits:
```
Comment (vgro) [inline docs/source/virtual_ecosystem/module_overview.md:231]:
```suggestion
Animal Model requires the following traits:
```
just for consistency ;-)

**Review by vgro — APPROVED** [review]
Looks good to me!

**Pull request merged**